python/tests/test_api.py

```diff
@@ -29,135 +29,408 @@ def setup_test_api():
     ),
+    pytest.param(
+        {"request": {"display": "hba1c 1/15/22 from outside lab"}},
+        "43396009",
+        id="display-request",
+    ),
+    pytest.param(
+        {
+            "request": [
+                {"code": "4548-4"},
+                {"display": "hba1c 1/15/22 from outside lab"},
+            ]
+        },
+        ["4548-4", "43396009"],
+        id="batch",
+    ),
 ]
 
 
 @pytest.mark.parametrize("payload, expected", _STANDARDIZE_LAB_PAYLOADS)
 def test_api_standardize_lab_should_standardize(payload, expected):
     response = TEST_API.standardize_lab(**payload)
+    positional_response = TEST_API.standardize_lab(*[payload[key] for key in payload])
 
+    assert response == positional_response
     assert response is not None
-    assert any(coding["code"] == expected for coding in response["coding"])
+    if isinstance(response, list):
+        assert len(response) == 2
+        assert any(coding["code"] == expected[0] for coding in response[0]["coding"])
+        assert any(coding["code"] == expected[1] for coding in response[1]["coding"])
+    else:
+        assert any(coding["code"] == expected for coding in response["coding"])
 
 
 _STANDARDIZE_MEDICATION_PAYOADS = [
     pytest.param({"code": "861004", "system": "RxNorm"}, "861004", id="rxnorm"),
+    pytest.param(
+        {"request": {"code": "861004", "system": "RxNorm"}},
+        "861004",
+        id="rxnorm-request",
+    ),
     pytest.param({"code": "59267-1000-02"}, "59267100002", id="ndc"),
+    pytest.param(
+        {"request": {"code": "59267-1000-02"}}, "59267100002", id="ndc-request"
+    ),
     pytest.param(
         {"display": "Jentadueto extended (linagliptin 2.5 / metFORMIN  1000mg)"},
         "1796093",
         id="display",
     ),
+    pytest.param(
+        {
+            "request": {
+                "display": "Jentadueto extended (linagliptin 2.5 / metFORMIN  1000mg)"
+            }
+        },
+        "1796093",
+        id="display-request",
+    ),
+    pytest.param(
+        {
+            "request": [
+                {"code": "861004", "system": "RxNorm"},
+                {"code": "59267-1000-02"},
+                {
+                    "display": "Jentadueto extended (linagliptin 2.5 / metFORMIN  1000mg)"
+                },
+            ]
+        },
+        ["861004", "59267100002", "1796093"],
+        id="batch",
+    ),
 ]
 
 
 @pytest.mark.parametrize("payload, expected", _STANDARDIZE_MEDICATION_PAYOADS)
 def test_api_standardize_medication_should_standardize(payload, expected):
     response = TEST_API.standardize_medication(**payload)
+    positional_response = TEST_API.standardize_medication(
+        *[payload[key] for key in payload]
+    )
 
+    assert response == positional_response
     assert response is not None
-    assert any(coding["code"] == expected for coding in response["coding"])
+    if isinstance(response, list):
```

Comment:
This is just so that the test can appropriately determine the right expected structure. I can split the Python tests to be batch and single, as I did in the TypeScript versions, if that makes it clearer.

In short, no, certainly not. The overloaded call will type the response for you appropriately.

![image](https://github.com/CareEvolution/OrchestrateSDK/assets/4954394/cdd5fb50-7425-4a48-a1b1-8f4c5923f880)

![image](https://github.com/CareEvolution/OrchestrateSDK/assets/4954394/20f74ae2-90c6-4682-a964-4d8eadd009c8)